service/src/main/java/bio/terra/workspace/amalgam/landingzone/azure/LandingZoneApiDispatch.java

```diff
@@ -76,23 +69,24 @@ public ApiAzureLandingZoneResult createAzureLandingZone(
             .version(body.getVersion())
             .parameters(
                 MapperUtils.LandingZoneMapper.landingZoneParametersFrom(body.getParameters()))
-            .landingZoneTarget(target)
+            .billingProfileId(body.getBillingProfileId())
             .build();
-    String jobId =
+    return toApiAzureLandingZoneResult(
```

Comment:
Kind of subtle, I ran into a concurrency issue here.

Before the code was doing this:
1. Call `LandingZoneService.startLandingZoneCreationJob`  --> returns a job UUID
2. Call `LandingZoneService getAsyncJobResult(jobId)` --> returns an AsyncJobResult

However now since both of these calls check Sam, step 2 would fail if the Sam resource hadn't been created yet (which happens asynchronously).

So I changed `LandingZoneService.startLandingZoneCreationJob` to just return an `AsyncJobResult` itself so we don't need the second call to fetch the async job result.